lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,203 @@
+    pub const LEVEL           : &str     = "level";
+
+    /// (optional) (deprecated) Send as additional field instead.
+    pub const FACILITY        : &str     = "facility";
+
+    /// (optional) (deprecated) The line in a file that caused the error (decimal). Send as additional field instead.
+    pub const LINE            : &str     = "line";
+
+    /// (optional) (deprecated) The file (with path if you want) that caused the error. Send as additional field instead.
+    pub const FILE            : &str     = "file";
+}
+pub use gelf_fields::*;
+
+/// Config used to build a `GelfDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct GelfDeserializerConfig;
+
+impl GelfDeserializerConfig {
+    /// Build the `GelfDeserializer` from this configuration.
+    pub fn build(&self) -> GelfDeserializer {
+        GelfDeserializer::default()
+    }
+
+    /// Return the type of event built by this deserializer.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            .with_field(VERSION, Kind::bytes(), None)
+            .with_field(HOST, Kind::bytes(), None)
+            .with_field(SHORT_MESSAGE, Kind::bytes(), None)
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
+            .optional_field(FILE, Kind::bytes(), None)
+            // Every field with an underscore (_) prefix will be treated as an additional field.
+            // Allowed characters in field names are any word character (letter, number, underscore), dashes and dots.
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
+        //
+    }
+}
+
+/// Deserializer that builds an `Event` from a byte frame containing a GELF log
+/// message.
+#[derive(Debug, Clone)]
+pub struct GelfDeserializer {
+    regex: Regex,
+}
+
+impl Default for GelfDeserializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GelfDeserializer {
+    /// Create a new GelfDeserializer
+    pub fn new() -> GelfDeserializer {
+        GelfDeserializer {
+            regex: Regex::new(r"^[\w\.\-]*$").unwrap(),
+        }
+    }
+
+    // Adds fields from parsed GelfMessage to the event, adhering to GELF spec
+    fn insert_fields_from_gelf(
+        &self,
+        event: &mut Event,
+        parsed: &GelfMessage,
+    ) -> vector_core::Result<()> {
+        let log = event.as_mut_log();
+
+        log.insert(VERSION, parsed.version.to_string());
```

Comment:
> What guarantees does GELF provide for forward compatibility?

Good question... I have search but so far I'm not seeing much online discussing that.
In their spec they specifically reference "1.1" and don't mention any other versions. It also appears that the version has not changed since 2013...
I can add an assertion on the version if that's the best approach for this kind of situation. We would then return an error if the version was not 1.1 ? Or do you mean panic the process?

Reply:
Yeah, I'm not finding much either. I'd suggest we be conservative and assert on `1.1`, returning a decoding error otherwise. If they followed semantic versioning I'd say we could allow `1.x` but I'm not finding a statement to that effect.

As you note, it seems unlikely to change 🙂

Reply:
Cool, it looks like the decoder logic above this maps the `vector_core::Error`s to a ParserError